src/main/java/graphql/annotations/processor/GraphQLAnnotations.java

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * A utility class for extracting GraphQL data structures from annotated
+ * elements.
+ */
+@Component
+public class GraphQLAnnotations implements GraphQLAnnotationsProcessor {
+
+    private GraphQLObjectHandler graphQLObjectHandler;
+    private ProcessingElementsContainer container;
+
+    private GraphQLObjectInfoRetriever graphQLObjectInfoRetriever;
+
+    public GraphQLAnnotations() {
+        this(new DefaultTypeFunction(new GraphQLInputProcessor(), new GraphQLOutputProcessor()), new GraphQLObjectInfoRetriever(), new GraphQLObjectHandler());
+    }
+
+    public GraphQLAnnotations(TypeFunction defaultTypeFunction, GraphQLObjectInfoRetriever graphQLObjectInfoRetriever, GraphQLObjectHandler graphQLObjectHandler) {
+        this.graphQLObjectInfoRetriever = graphQLObjectInfoRetriever;
+        this.defaultTypeFunction = defaultTypeFunction;
+        this.graphQLObjectHandler = graphQLObjectHandler;
+        this.container = initializeContainer(this.defaultTypeFunction);
+    }
+
+    private ProcessingElementsContainer initializeContainer(TypeFunction defaultTypeFunction) {
+        Map<String, graphql.schema.GraphQLType> typeRegistry = new HashMap<>();
+        Map<Class<?>, Set<Class<?>>> extensionsTypeRegistry = new HashMap<>();
+        final Stack<String> processing = new Stack<>();
+        Relay relay = new Relay();
+        ProcessingElementsContainer container = new ProcessingElementsContainer(defaultTypeFunction, relay, typeRegistry, extensionsTypeRegistry, processing);
+        return container;
+    }
+
+    public static GraphQLAnnotations instance = new GraphQLAnnotations();
+
+    public static GraphQLAnnotations getInstance() {
+        return instance;
+    }
+
+    public void setRelay(Relay relay) {
+        this.container.setRelay(relay);
+    }
+
+
+    public String getTypeName(Class<?> objectClass) {
+        GraphQLName name = objectClass.getAnnotation(GraphQLName.class);
+        return toGraphqlName(name == null ? objectClass.getSimpleName() : name.value());
+    }
+
+    public static GraphQLObjectType object(Class<?> object) throws GraphQLAnnotationsException {
+        return new GraphQLObjectHandler().getObject(object, getInstance().getContainer());
+    }
+
+    public static class GraphQLFieldDefinitionWrapper extends GraphQLFieldDefinition {
+
+        public GraphQLFieldDefinitionWrapper(GraphQLFieldDefinition fieldDefinition) {
+            super(fieldDefinition.getName(), fieldDefinition.getDescription(), fieldDefinition.getType(),
+                    fieldDefinition.getDataFetcher(), fieldDefinition.getArguments(), fieldDefinition.getDeprecationReason());
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return obj instanceof GraphQLFieldDefinition &&
+                    ((GraphQLFieldDefinition) obj).getName().contentEquals(getName());
+        }
+    }
+
+    protected TypeFunction defaultTypeFunction;
```

Comment:
It is still needed to have a :  ` @Reference(target = "(type=default)") public void setDefaultTypeFunction(TypeFunction function)` method, in order to register default type function in OSGi. OSGi framework will always use `GraphQLAnnotations()` constructor (the other one cannot be used), and will then call `@Reference` setter to fill dependencies, with the instance of DefaultTypeFunction it has created. 
By the way, if we assume that we will always use a DefaultTypeFunction (you actually assume that in registerType method), it's also fine for me - then you can change TypeFunction to DefaultTypeFunction. But the setDefaultTypeFunction() with `@Reference` is still needed as we want the instance created by OSGi to be injected here.

Reply:
Ive added this method but im not sure its what you want.
You are welcome to PR a support for OSGI because we do not understand this framework